app/src/main/kotlin/com/wire/android/ui/home/conversations/selfdeletion/SelfDeletionMenuItems.kt

```diff
@@ -0,0 +1,118 @@
+import com.wire.android.ui.home.conversations.edit.OnComplete
+import com.wire.android.ui.home.messagecomposer.state.SelfDeletionDuration
+import com.wire.android.ui.theme.wireTypography
+
+@Composable
+fun SelfDeletionMenuItems(
+    currentlySelected: SelfDeletionDuration,
+    hideEditMessageMenu: (OnComplete) -> Unit,
+    onSelfDeletionDurationChanged: (SelfDeletionDuration) -> Unit,
+): List<@Composable () -> Unit> {
+
+    val onSelfDeletionDurationSelected: (SelfDeletionDuration) -> Unit = { selfDeleteDuration ->
+        hideEditMessageMenu {
+            onSelfDeletionDurationChanged(selfDeleteDuration)
+        }
+    }
+
+    return buildList {
+        add {
+            val duration = SelfDeletionDuration.None
+            SelfDeletionDurationMenuItem(
+                duration = duration,
+                isSelected = currentlySelected == duration,
+                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
+            )
+        }
+        add {
+            val duration = SelfDeletionDuration.TenSeconds
+            SelfDeletionDurationMenuItem(
+                duration = duration,
+                isSelected = currentlySelected == duration,
+                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
+            )
+        }
+        add {
+            val duration = SelfDeletionDuration.FiveMinutes
+            SelfDeletionDurationMenuItem(
+                duration = duration,
+                isSelected = currentlySelected == duration,
+                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
+            )
+        }
+        add {
+            val duration = SelfDeletionDuration.OneHour
+            SelfDeletionDurationMenuItem(
+                duration = duration,
+                isSelected = currentlySelected == duration,
+                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
+            )
+        }
+        add {
+            val duration = SelfDeletionDuration.OneDay
+            SelfDeletionDurationMenuItem(
+                duration = SelfDeletionDuration.OneDay,
+                isSelected = currentlySelected == duration,
+                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
+            )
+        }
+        add {
+            val duration = SelfDeletionDuration.OneWeek
+            SelfDeletionDurationMenuItem(
+                duration = SelfDeletionDuration.OneWeek,
+                isSelected = currentlySelected == duration,
+                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
+            )
+        }
+        add {
+            val duration = SelfDeletionDuration.FourWeeks
+            SelfDeletionDurationMenuItem(
+                duration = SelfDeletionDuration.FourWeeks,
+                isSelected = currentlySelected == duration,
+                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
+            )
+        }
+    }
```

Comment:
It's a copy-pasted code which differs only by `duration` - it could be extracted to a method and re-used 😉

Reply:
but how ? 🤔 we need to add stuff 7 times so we need to call it 7 times

Reply:
Maybe extracting to a separate method doesn't make much sense, but I'm thinking maybe something like this:
```
    return SelfDeletionDuration.values().map { duration ->
        {
            SelfDeletionDurationMenuItem(
                duration = duration,
                isSelected = currentlySelected == duration,
                onSelfDeletionDurationSelected = onSelfDeletionDurationSelected
            )
        }
    }
```

Reply:
somehow I did not think about 😮‍💨 thanks